### PR TITLE
RD-1293 Added support for TS in widget backend

### PR DESCRIPTION
**Pull request by kubama**
## Description
Added support for TS in widget backend

## Screenshots / Videos
N/A

## Checklist

- [X] My code follows the UI Code Style.
- [x] I verified that all tests and checks have passed.
- [X] I verified if that change requires a documentation update.
- [X] I added proper labels to this PR.

## Tests
https://jenkins.cloudify.co/view/UI/job/Stage-UI-System-Test/1971/

 

## Documentation
I'll update https://docs.cloudify.co/latest/developer/writing_widgets/widget-backend/ in separate PR

### CODE REVIEW

File: backend/package.json [hunk 1]
```diff
@@ -29,6 +29,7 @@
     "delay": 2000
   },
   "dependencies": {
+    "@tsconfig/node12": "^1.0.9",
```
Comment (qooban) [inline backend/package.json:32]: Maybe worth to use node v14 (as that is our current version we support)?

**Review by qooban — COMMENTED** [review]
Added one comment.
As I'll be not available till the end of this week, please don't wait for my approval.

**Review by jawamartus — APPROVED** [review]
LGFM

File: backend/package.json (hunk 1 above)
Comment (kubama) [reply backend/package.json]: Done

File: backend/package.json [hunk 2]
```diff
@@ -29,6 +29,7 @@
     "delay": 2000
   },
   "dependencies": {
+    "@tsconfig/node14": "^1.0.1",
```
Comment (jawamartus) [inline backend/package.json:32]: What about package-lock.json? Do I think correctly that we should update it as well?

**Review by jawamartus — APPROVED** [review]
LGFM with exception for one doubt about package-lock.json

File: backend/package.json (hunk 2 above)
Comment (kubama) [reply backend/package.json:32]: I used `npm i` for adding this dependency so I assume `package-lock.json` would be updated if it was required.
Also, this package was already present as `ts-node` dependency (I added it as it is now used directly in an `import` statement) so I think all is good here.

**Pull request merged**